File: src/main/java/com/neo/caption/ocr/domain/entity/CaptionRow.java
package com.neo.caption.ocr.domain.entity;

import lombok.AllArgsConstructor;
import lombok.Data;

@Data
@AllArgsConstructor
public class CaptionRow implements Cloneable {

    private int startIndex;
    private int endIndex;
    private String caption;

    @Override
    public CaptionRow clone() {
        try {
            return (CaptionRow) super.clone();
        } catch (CloneNotSupportedException e) {
            throw new AssertionError();
        }
    }
}

File: src/main/kotlin/com/neo/caption/ocr/domain/entity/CaptionRow.kt
package com.neo.caption.ocr.domain.entity

import org.opencv.core.Mat

data class CaptionRow(
    var start: Int,
    var end: Int,
    val mat: Mat,
    var caption: String
) {
    fun clone() = this.copy(mat = Mat().also { this.mat.copyTo(it) })
}